lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -266,6 +266,12 @@ def install_puppet_on(hosts, opts={})
               if host['platform'] =~ /el-(5|6|7)/
                 relver = $1
                 install_puppet_from_rpm_on(host, opts.merge(:release => relver, :family => 'el'))
+              elsif host['platform'] =~ /sles-(11|12)/
+                relver = $1
+                # Need to get rid of whatever Puppet might have shipped on the
+                # upstream images
+                host.uninstall_package('puppet')
```

Comment:
@kevpl Yeah, the official LEAP images have Puppet and a bunch of other things installed so we need to rip them off first. Also, this step doesn't hurt since we're trying to install the "right" version.

The SUSE packages don't differentiate properly between the different versions of Puppet and will conflict.